File: codeSamples/Python/vulnerable_sample.py
import os
import subprocess
import pickle


# Hallazgo: Nombre de clase corto
class a:
    # Hallazgo: Secreto en el código
    api_key = "SECRET_KEY_12345"

    # Hallazgo: Nombre de función corto
    def r(self, cmd):
        # Hallazgo: Llamada peligrosa
        os.system(cmd) 
        
        # Hallazgo: Llamada peligrosa (especialmente con shell=True)
        subprocess.run("ls -l", shell=True) 

# Hallazgo: Bloque except vacío/inútil
try:
    # Algo que podría fallar
    data = {'key': 'value'}
    pickle.loads(data) # Esto dará error, pero lo ignoramos
except:
    pass

# Hallazgo: Punto de entrada principal
if __name__ == "__main__":
    app = a()
    # Hallazgo: Acceso a ruta sensible
    user_home = "/home/user"
    app.r("echo 'hello'")
    os.path.abspath(__file__)

File: codeSamples/JavaScript/vulnerable_sample.js
// Importación sospechosa (Node.js)
import * as fs from 'fs';
import { exec } from 'child_process'; // Llamada peligrosa

// Hallazgo: Nombre de clase corto
class A {
    // Hallazgo: Secreto en el código
    constructor() {
        this.apiKey = "JS_SECRET_API_KEY";
    }

    // Hallazgo: Nombre de método corto
    m() {
        // Hallazgo: Llamada peligrosa
        eval("console.log('Malicious code');"); 
        exec('rm -rf /', (err, stdout, stderr) => {});
    }

    // Hallazgo: Método con cuerpo vacío (override sospechoso)
    // Aunque no hay @override en JS, un método vacío es sospechoso
    doNothing() {} 
}

// Hallazgo: Función declarada globalmente con nombre corto
function f() {
    console.log("Short function name");
}

// Hallazgo: Bloque catch vacío
try {
    throw new Error("Test error");
} catch (e) {
    // EMPTY_CATCH_BLOCK
}

// Hallazgo: Acceso a ruta sensible
const sensitivePath = "/etc/passwd";
console.log(sensitivePath);

// Hallazgo: Código auto-consciente
console.log(__dirname); // Node.js global
console.log(process.argv[0]); // Node.js

// Punto de entrada (proxy)
function main() {
    let obj = new A();
    obj.m();
    f();
}
main();